lighthouse-core/audits/byte-efficiency/unused-javascript.js

```diff
@@ -71,183 +72,61 @@ class UnusedJavaScript extends ByteEfficiencyAudit {
-          const key = mapping.sourceURL || '(unmapped)';
-          files[key] = (files[key] || 0) + 1;
-        }
-        offset += 1;
-      }
-    }
-
-    const transferRatio = lengths.transfer / lengths.content;
-    const topUnusedFilesSizes = Object.entries(files)
-      .filter(([_, unusedBytes]) => unusedBytes * transferRatio >= bundleSourceUnusedThreshold)
-      .sort(([_, unusedBytes1], [__, unusedBytes2]) => unusedBytes2 - unusedBytes1)
-      .slice(0, 5)
-      .map(([key, unusedBytes]) => {
-        const total = key === '(unmapped)' ? bundle.sizes.unmappedBytes : bundle.sizes.files[key];
-        return {
-          key,
-          unused: Math.round(unusedBytes * transferRatio),
-          total: Math.round(total * transferRatio),
-        };
-      });
-
-    const commonSourcePrefix = commonPrefix([...bundle.map._sourceInfos.keys()]);
-    Object.assign(item, {
-      sources: trimCommonPrefix(topUnusedFilesSizes.map(d => d.key), commonSourcePrefix),
-      sourceBytes: topUnusedFilesSizes.map(d => d.total),
-      sourceWastedBytes: topUnusedFilesSizes.map(d => d.unused),
-    });
-  }
-
-  /**
-   * @param {WasteData[]} wasteData
-   * @param {string} url
-   * @param {ReturnType<typeof UnusedJavaScript.determineLengths>} lengths
-   * @return {LH.Audit.ByteEfficiencyItem}
-   */
-  static mergeWaste(wasteData, url, lengths) {
-    let unused = 0;
-    let content = 0;
-    // TODO: this is right for multiple script tags in an HTML document,
-    // but may be wrong for multiple frames using the same script resource.
-    for (const usage of wasteData) {
-      unused += usage.unusedLength;
-      content += usage.contentLength;
-    }
-
-    const wastedRatio = (unused / content) || 0;
-    const wastedBytes = Math.round(lengths.transfer * wastedRatio);
-
-    return {
-      url: url,
-      totalBytes: lengths.transfer,
-      wastedBytes,
-      wastedPercent: 100 * wastedRatio,
-    };
-  }
-
-  /**
-   * @param {WasteData[]} wasteData
-   * @param {LH.Artifacts.NetworkRequest} networkRecord
-   */
-  static determineLengths(wasteData, networkRecord) {
-    let unused = 0;
-    let content = 0;
-    // TODO: this is right for multiple script tags in an HTML document,
-    // but may be wrong for multiple frames using the same script resource.
-    for (const usage of wasteData) {
-      unused += usage.unusedLength;
-      content += usage.contentLength;
-    }
-    const transfer = ByteEfficiencyAudit.estimateTransferSize(networkRecord, content, 'Script');
-
-    return {
-      content,
-      unused,
-      transfer,
-    };
-  }
-
   /**
    * @param {LH.Artifacts} artifacts
    * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
    * @param {LH.Audit.Context} context
    * @return {Promise<ByteEfficiencyAudit.ByteEfficiencyProduct>}
    */
   static async audit_(artifacts, networkRecords, context) {
-    const bundles = artifacts.SourceMaps ? await JSBundles.request(artifacts, context) : [];
```

Comment:
oof, yeah not sure how I missed this. I guess I was still in "oh just code movement" mode until the additions below 🤦